app/hooks/useToggle.js

```diff
@@ -23,8 +23,14 @@ export function useToggle(navigation, headerIconName, initialState = false) {
     })
   }, [navigation]);
 
+  const toggle = () => setIsOn(!isOn);
   const turnOn = () => setIsOn(true);
   const turnOff = () => setIsOn(false);
```

Comment:
This might be a good place to memoize [using `useCallback`](https://reactjs.org/docs/hooks-reference.html#usecallback), since currently you're basically creating three new functions on every render that uses this hook.